### PR TITLE
Server - User model

### LINKED ISSUE #5: Server - User model

### PR DESCRIPTION
**Pull request by OfirArbel**
fix #5

### CODE REVIEW

File: server/api/user/userController.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        });
+      } else {
+        req.user = user;
+        next();
+        }
+      });
+};
+
+exports.createNew = function(req, res){
+  User.create(req.body).
```
Comment (nirgn975) [inline server/api/user/userController.js]: If there is an error when creating the user? Why you don't handle it?

File: server/api/user/userController.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+const bodyParser = require('body-parser')
+
+const User = require('./userModel');
+
+exports.params = function(req, res, next){
+  User.findOne({'name': req.params['name']})
```
Comment (nirgn975) [inline server/api/user/userController.js]: You search by `name`. So `name` is unique in the system? What about 2 users with the name `Ofir`, which one do you return?

File: server/api/user/userModel.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const mongoose = require('mongoose');
+
+const UserSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
+    unique: true
```
Comment (nirgn975) [inline server/api/user/userModel.js]: So there is no option in your system to register a new user named `Ofir` if there is already one. You're limiting your sisters income..

File: server/api/user/userModel.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  age: {
+    type: Number,
+    required: true,
+    unique: false,
+  },
+  tell: {
```
Comment (nirgn975) [inline server/api/user/userModel.js]: What the hell is `tell`? you mean `phone` or `phoneNumber`?

File: server/api/user/userModel.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+const UserSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
+    unique: true
+  },
+  age: {
+    type: Number,
+    required: true,
+    unique: false,
```
Comment (nirgn975) [inline server/api/user/userModel.js]: I think the default option is `unique: false` so I don't think you need to specify it. Check it.

File: server/api/user/userModel.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  },
+  tell: {
+    type: String,
+    required: true,
+    unique: true
+  },
+},
+{
+  versionKey: false
```
Comment (nirgn975) [inline server/api/user/userModel.js]: What is this?

File: server/api/user/userController.js (hunk 2 above)
Comment (OfirArbel) [reply server/api/user/userController.js]: The name is unique

File: server/api/user/userModel.js (hunk 3 above)
Comment (OfirArbel) [reply server/api/user/userModel.js]: haha

File: server/api/user/userModel.js (hunk 4 above)
Comment (OfirArbel) [reply server/api/user/userModel.js]: yes :\.... I will change

File: server/api/user/userModel.js (hunk 6 above)
Comment (OfirArbel) [reply server/api/user/userModel.js]: Without this definition, mongoose saves the document with ```_v``` key. This key value contains the internal revision of the document. I don't want this.

File: server/api/user/userController.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+const User = require('./userModel');
+
+exports.params = (req, res, next) => {
+  User.findOne({'name': req.params['name']})
+    .then(function(user){
```
Comment (nirgn975) [inline server/api/user/userController.js]: What about _anonymous function_?

File: server/api/user/userController.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+      } else {
+        req.user = user;
+        next();
+        }
+      });
+};
+
+exports.createNew = (req, res) => {
+  User.create(req.body)
+    .then(function(User){
```
Comment (nirgn975) [inline server/api/user/userController.js]: What about _anonymous function_?

File: server/api/user/userController.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      res.json({
+        _message: "User successfully created!",
+        user: User
+      });
+  });
+};
+
+exports.getUser = (req, res, next) => {
+  var user = req.user;
```
Comment (nirgn975) [inline server/api/user/userController.js]: Why are you using `var`?

File: server/api/user/userController.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+  res.json({
+    _message: "User fetched successfully!",
+    user: user
+  });
+};
+
+exports.getAll = (req, res) => {
+  User.find({})
+    .then(function(users){
```
Comment (nirgn975) [inline server/api/user/userController.js]: What about _anonymous function_?

File: server/api/user/userModel.js (hunk 3 above)
Comment (nirgn975) [reply server/api/user/userModel.js]: I'm not kidding. Do you think the `name` of the user should be `unique`? Why?

File: server/api/user/userModel.js (hunk 6 above)
Comment (nirgn975) [reply server/api/user/userModel.js]: Why you don't want it?

File: server/api/user/userModel.js [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    type: Number,
+    required: true,
+  },
+  phoneNumber: {
+    type: String,
+    required: true,
+    unique: true
+  },
+},
+{
```
Comment (nirgn975) [inline server/api/user/userModel.js]: This should be in the same line, like:
```
}, {
```

File: server/api/user/userModel.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+  versionKey: false
+}
+);
```
Comment (nirgn975) [inline server/api/user/userModel.js]: This should be on the line before, like:
```
});
```

File: server/api/user/userModel.js (hunk 3 above)
Comment (OfirArbel) [reply server/api/user/userModel.js]: U WIM!@#@#!@!$# I added email field

**Pull request merged**